hc_dpki/src/seed.rs

```diff
@@ -0,0 +1,558 @@
+    ///
+    /// @param {string} passphrase - the decryption passphrase
+    ///
+    /// @return Type FromBundle
+    pub fn from_seed_bundle(
+        bundle: bundle::KeyBundle,
+        passphrase: String,
+        config: Option<PwHashConfig>,
+    ) -> Result<FromBundle, HolochainError> {
+        let mut passphrase = SecBuf::with_insecure_from_string(passphrase);
+
+        let seed_data_decoded = base64::decode(&bundle.data).unwrap();
+        let seed_data_string = str::from_utf8(&seed_data_decoded).unwrap();
+
+        let seed_data_deserialized: bundle::ReturnBundleData =
+            json::decode(&seed_data_string).unwrap();
+        let mut seed_data = SecBuf::with_secure(32);
+
+        util::pw_dec(
+            &seed_data_deserialized,
+            &mut passphrase,
+            &mut seed_data,
+            config,
+        )?;
+
+        match bundle.bundle_type.as_ref() {
+            "hcRootSeed" => Ok(FromBundle::Rs(RootSeed::new(seed_data))),
+            "hcDeviceSeed" => Ok(FromBundle::Ds(DeviceSeed::new(seed_data))),
+            "hcDevicePinSeed" => Ok(FromBundle::Dps(DevicePinSeed::new(seed_data))),
+            _ => Err(HolochainError::new(&format!("Invalid Bundle type"))),
+        }
+    }
+
+    ///  generate a persistence bundle with hint info
+    ///
+    ///  @param {string} passphrase - the encryption passphrase
+    ///
+    ///  @param {string} hint - additional info / description for persistence
+    ///
+    /// @return {KeyBundle} - bundle of the seed
+    pub fn get_seed_bundle(
+        &mut self,
+        passphrase: String,
+        hint: String,
+        config: Option<PwHashConfig>,
+    ) -> Result<bundle::KeyBundle, HolochainError> {
+        let mut passphrase = SecBuf::with_insecure_from_string(passphrase);
+        let seed_data: bundle::ReturnBundleData =
+            util::pw_enc(&mut self.seed_buf, &mut passphrase, config)?;
+
+        // convert -> to string -> to base64
+        let seed_data_serialized = json::encode(&seed_data).unwrap();
+        let seed_data_encoded = base64::encode(&seed_data_serialized);
+
+        Ok(bundle::KeyBundle {
+            bundle_type: self.seed_type.clone(),
+            hint,
+            data: seed_data_encoded,
+        })
+    }
+
+    ///  Initialize this seed class with persistence bundle type and private seed
+    ///
+    ///  @param {string} stype - the persistence bundle type
+    ///
+    ///  @param {SecBuf|string} seed - the private seed data (as a buffer or mnemonic)
+    pub fn new(stype: &String, sm: InitializeSeed) -> Self {
+        match sm {
+            SeedInit(s) => Seed {
+                seed_type: stype.clone(),
+                seed_buf: s,
+            },
+            MnemonicInit(phrase) => {
+                let mnemonic = Mnemonic::from_phrase(phrase, Language::English).unwrap();
+                let entropy: &[u8] = mnemonic.entropy();
+                let mut buf = SecBuf::with_insecure(entropy.len());
```

Comment:
`secure` here.

plus... erm, is entropy here immutable? *shudder* we might not have a way of zeroing this secret in memory. See if there's any way you can clear it out, otherwise add a comment to come back and address it... we might have to make our own mnemonic parser.

Reply:
I am working on this. But I have put a comment for now.